crates/router/src/core/revenue_recovery/types.rs

```diff
@@ -322,18 +389,29 @@ impl Decision {
         state: &SessionState,
         intent_status: enums::IntentStatus,
         called_connector: enums::PaymentConnectorTransmission,
-        active_attempt_id: Option<id_type::GlobalAttemptId>,
+        active_attempt_id: Option<&id_type::GlobalAttemptId>,
         revenue_recovery_data: &storage::revenue_recovery::RevenueRecoveryPaymentData,
         payment_id: &id_type::GlobalPaymentId,
     ) -> RecoveryResult<Self> {
         logger::info!("Entering get_decision_based_on_params");
 
+        let active_attempt_id = revenue_recovery_data
```

Comment:
Explain here, how there is two source of truth for the same data? and why there should be two source of truth?